fastapi_pokedex_api/pokeapi_requests.py

```diff
@@ -1,13 +1,46 @@
-from fastapi import FastAPI
+import fastapi
+import pydantic.error_wrappers
+import requests
+from pydantic import typing
 
-from fastapi_pokedex_api import pokedex_schema
-from fastapi_pokedex_api import api_app
+from fastapi_pokedex_api.configs.pokeapi_config import PokeapiSettings
+from fastapi_pokedex_api.pokedex_schema import PokedexBase
 
-app = FastAPI()
 
+def get_pokedata(ip_settings: PokeapiSettings, poke_name: str) -> PokedexBase:
+    name = poke_name.lower()
+    try:
+        response = requests.get("{}{}".format(ip_settings.POKEAPI_IP, name))
+        response.raise_for_status()
+    except requests.exceptions.HTTPError:
+        raise fastapi.HTTPException(
+            status_code=404,
+            detail="pokemon '{}' not found. Check spelling".format(name),
+        )
 
-@app.get(
-    "/pokedex/{poke_name}", response_model=pokedex_schema.PokedexBase, tags=["Pokemon"]
-)
-async def get_pokedata(poke_name: str) -> pokedex_schema.PokedexBase:
-    return api_app.get_pokedata(poke_name=poke_name)
+    try:
+        details = response.json()
+    except requests.exceptions.JSONDecodeError:
+        raise fastapi.HTTPException(status_code=406, detail="response not jsonable")
+
+    try:
+        return PokedexBase(
+            name=name,
+            description=english_description(details.get("flavor_text_entries", {})),
+            habitat=details.get("habitat").get("name")
+            if details.get("habitat") is not None
+            else None,
```

Comment:
This can be simplified to

```
details.get("habitat", {}).get("name")
```

`Dictionary.get()` takes a second parameter which is the **default**. Instead of `None`, you can return an empty dict which the next part can call `.get` on (this time returning the default `None` value.

Reply:
I tried this before but details.get("habitat", {}) returns None for some pokemon (such as piplup) instead of the "habitat" value or an empty dict. 
This means that if I call another .get() method on it an AtributeError is raised as you NoneType has no attribute .get() .
That's why I had to keep with the if else statments

Reply:
I think that because details["habitat"] = None (or null) for pokemon such as piplup, instead of "habitat" just being a missing key, that's why details.get("habitat", {}) returns None instead of {}